interlocking/infrastructureprovider/infrastructureprovider.py

```diff
@@ -1,18 +1,53 @@
 from abc import ABC, abstractmethod
-import asyncio
+from yaramo.model import Node, Signal
+
 
 class InfrastructureProvider(ABC):
 
-    def __init__(self):
+    def __init__(self,
+                 only_apply_for_signals: list[str] = None,
+                 only_apply_for_points: list[str] = None,
+                 apply_for_all_signals_except: list[str] = None,
+                 apply_for_all_points_except: list[str] = None):
+        if only_apply_for_signals is None:
+            only_apply_for_signals = []
+        self.only_apply_for_signals = only_apply_for_signals
+        if only_apply_for_points is None:
+            only_apply_for_points = []
+        self.only_apply_for_points = only_apply_for_points
+        if apply_for_all_signals_except is None:
+            apply_for_all_signals_except = []
+        self.apply_for_all_signals_except = apply_for_all_signals_except
+        if apply_for_all_points_except is None:
+            apply_for_all_points_except = []
+        self.apply_for_all_points_except = apply_for_all_points_except
+
+        if len(self.only_apply_for_signals) > 0 and len(self.apply_for_all_signals_except) > 0:
+            raise Exception(f"You can not limit the infrastructure provider with only_apply_for_signals and "
+                            f"apply_for_all_signals_except at the same time.")
+        if len(self.only_apply_for_points) > 0 and len(self.apply_for_all_points_except) > 0:
+            raise Exception(f"You can not limit the infrastructure provider with only_apply_for_points and "
+                            f"apply_for_all_points_except at the same time.")
+
         self.tds_count_in_callback = None
         self.tds_count_out_callback = None
 
     #
     # Point Interaction
     #
 
+    async def call_turn_point(self, yaramo_point: Node, target_orientation: str):
+        point_id = yaramo_point.uuid[-5:]
+        if point_id in self.only_apply_for_points:
+            return await self.turn_point(yaramo_point, target_orientation)
+        if len(self.only_apply_for_points) == 0 and point_id not in self.apply_for_all_points_except:
+            return await self.turn_point(yaramo_point, target_orientation)
+
+        # return True to skip this call and not prevent successfully turning of point.
+        return True
```

Comment:
How can we ensure that an infrastructure element has been handeled and not filtered by all Infrastructureproviders. Would it be an idea return False or raise an Exception, to force add an default Infrastructure provider.

Reply:
Hmm I think, it could be a feature, that you can just add an infrastructure provider for a selected set of elements. Nevertheless I think two options were possible:
- Print a warning for all elements that are not covered by an infrastructure provider
- And/or add a logging infrastructure provider for all of them.

We can also add a [setting](https://github.com/simulate-digital-rail/interlocking/blob/main/interlocking/model/helper/settings.py) with a default infrastructure provider for the last one. Do you understand?

Reply:
Agree.

Reply:
@dfriedenberger I've added that functionality. Please review again.